course6/toyapp_alexs/displaylib/transformation3d.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include "vector3d.h"
+
+namespace Display {
+
+class Transformation3D {
+public:
+    Transformation3D() {};
+    virtual Vector3D Apply(Vector3D vertex) const = 0;
+    virtual ~Transformation3D() = default;
+};
+
+class Move3D: public Transformation3D {
+public:
+    Move3D(Vector3D move);
+    Vector3D Apply(Vector3D vertex) const;
+private:
+    Vector3D mMove;
+};
+
+class Scale3D: public Transformation3D {
+public:
+    Scale3D(Vector3D scale);
+    Vector3D Apply(Vector3D vertex) const;
+private:
+    Vector3D mScale;
+};
+
+class Rotate3D: public Transformation3D {
+public:
+    Rotate3D(double thetaX, double thetaY, double thetaZ);
+    Vector3D Apply(Vector3D vertex) const;
+private:
+    double mThetaX;
+    double mThetaY;
+    double mThetaZ;
+
+    Vector3D ApplyX(Vector3D vertex) const;
+    Vector3D ApplyY(Vector3D vertex) const;
+    Vector3D ApplyZ(Vector3D vertex) const;
+};
```

Comment:
The entire idea behind transformation matrices is that you don't need to implement every operation separately.
You just construct a matrix representing the transformation, and you multiply vector by the matrix.
You could also combine transformation by multiplying matrices.

Reply:
will do this a bit later